sim/core/item_swaps.go

```diff
@@ -0,0 +1,170 @@
+package core
+
+import (
+	"time"
+
+	"github.com/wowsims/wotlk/sim/core/proto"
+)
+
+type OnSwapItem func(*Simulation)
+
+var onSwapCallbacks = map[int32]OnSwapItem{}
```

Comment:
Move this into the `ItemSwap` struct and make it a list instead of a map (no ID key). Then just call all swap functions registered to the character.